--- a/src/config.h	2008-07-21 10:56:45.000000000 -0600
+++ b/src/config.h	2008-07-21 10:59:39.000000000 -0600
@@ -13,13 +13,13 @@
 #endif
 
 // Configure as a coreboot payload.
-#define CONFIG_COREBOOT 0
+#define CONFIG_COREBOOT 1
 
 // Control how verbose debug output is.
-#define CONFIG_DEBUG_LEVEL 1
+#define CONFIG_DEBUG_LEVEL 6
 
 // Send debugging information to serial port
-#define CONFIG_DEBUG_SERIAL 0
+#define CONFIG_DEBUG_SERIAL 1
 
 // Support for int13 floppy drive access
 #define CONFIG_FLOPPY_SUPPORT 1
@@ -36,7 +36,7 @@
 // Support int 1a/b1 PCI BIOS calls
 #define CONFIG_PCIBIOS 1
 // Maximum number of PCI busses.
-#define CONFIG_PCI_BUS_COUNT 2
+#define CONFIG_PCI_BUS_COUNT 6
 // Support int 15/53 APM BIOS calls
 #define CONFIG_APMBIOS 1
 // Support int 19/18 system bootup support
