Index: etherboot-5.4.3/src/include/etherboot.h
===================================================================
--- etherboot-5.4.3.orig/src/include/etherboot.h	2007-04-23 20:22:10.000000000 -0600
+++ etherboot-5.4.3/src/include/etherboot.h	2007-04-23 20:22:23.000000000 -0600
@@ -327,7 +327,9 @@
 
 /* basemem.c */
 extern uint32_t get_free_base_memory ( void );
+#ifndef LINUXBIOS
 extern void allot_real_mode_stack ( void );
+#endif
 extern void forget_real_mode_stack ( void );
 extern void * allot_base_memory ( size_t );
 extern void forget_base_memory ( void*, size_t );
