diff -Naur qemu-0.9.0-orig/hw/pc.c qemu-0.9.0/hw/pc.c
--- qemu-0.9.0-orig/hw/pc.c	2007-07-29 19:34:43.000000000 -0300
+++ qemu-0.9.0/hw/pc.c	2007-07-29 19:39:48.000000000 -0300
@@ -481,13 +481,13 @@
 
     /* BIOS load */
     bios_offset = ram_size + vga_ram_size;
-    vga_bios_offset = bios_offset + 256 * 1024;
+    vga_bios_offset = bios_offset + 2048 * 1024;
 
     snprintf(buf, sizeof(buf), "%s/%s", bios_dir, BIOS_FILENAME);
     bios_size = get_image_size(buf);
     if (bios_size <= 0 || 
         (bios_size % 65536) != 0 ||
-        bios_size > (256 * 1024)) {
+        bios_size > (2048 * 1024)) {
         goto bios_error;
     }
     ret = load_image(buf, phys_ram_base + bios_offset);
diff -Naur qemu-0.9.0-orig/vl.h qemu-0.9.0/vl.h
--- qemu-0.9.0-orig/vl.h	2007-07-29 19:34:43.000000000 -0300
+++ qemu-0.9.0/vl.h	2007-07-29 19:40:19.000000000 -0300
@@ -170,7 +170,7 @@
 #elif defined(TARGET_MIPS)
 #define BIOS_SIZE (4 * 1024 * 1024)
 #else
-#define BIOS_SIZE ((256 + 64) * 1024)
+#define BIOS_SIZE ((2048 + 64) * 1024)
 #endif
 
 /* keyboard/mouse support */
