Index: svn/targets/gigabyte/ga_2761gxdk/Config.lb
===================================================================
--- svn.orig/targets/gigabyte/ga_2761gxdk/Config.lb
+++ svn/targets/gigabyte/ga_2761gxdk/Config.lb
@@ -42,7 +42,8 @@ romimage "normal"
 #	option ROM_IMAGE_SIZE=0x15800
 	option XIP_ROM_SIZE=0x40000
 	option LINUXBIOS_EXTRA_VERSION="$(shell cat ../../VERSION)_Normal"
-        payload ../../../../payloads/filo_uda1.elf
+#        payload ../../../../payloads/filo_uda1.elf
+	payload ../payload.elf
 end
 
 romimage "fallback" 
@@ -55,7 +56,8 @@ romimage "fallback" 
 #	option ROM_IMAGE_SIZE=0x15800
 	option XIP_ROM_SIZE=0x40000
 	option LINUXBIOS_EXTRA_VERSION="$(shell cat ../../VERSION)_Fallback"
-        payload ../../../../payloads/filo_uda1.elf
+#        payload ../../../../payloads/filo_uda1.elf
+	payload ../payload.elf
 end
 
 romimage "failover"
