Index: svn/arch/x86/stage0_i586.S
===================================================================
--- svn/arch/x86/stage0_i586.S	(revision 539)
+++ svn/arch/x86/stage0_i586.S	(working copy)
@@ -56,7 +56,8 @@
 	 * the ld hackery and other things. So leave it as is with this comment. 
 	 */
 
-	data32	lgdt %cs:gdtptr
+	movl	$gdtptr, %ebx
+	lgdt	%cs:(%bx)
 
 	movl	%cr0, %eax
 	andl	$0x7FFAFFD1, %eax /* PG,AM,WP,NE,TS,EM,MP = 0 */
