diff --git a/src/southbridge/amd/cs5536/cs5536.c b/src/southbridge/amd/cs5536/cs5536.c
index 3e6f96b..6c0c8c0 100755
--- a/src/southbridge/amd/cs5536/cs5536.c
+++ b/src/southbridge/amd/cs5536/cs5536.c
@@ -90,6 +90,29 @@ static unsigned char ec_inb(unsigned sho
     return inb(EC_INDEX(3));
 }
 
+#define EC_3920_SIGNATURE  0x09
+
+static unsigned char eccmd(unsigned char command) {
+
+	unsigned char ret;
+
+	if ((ret = inb(0x6c)) & 1)
+		ret = inb(0x68);
+
+	/* Write the command */
+	outb(command, 0x6C);
+
+	/* Wait for the EC response */
+	while((inb(0x6C) & 3) != 1);
+
+	/* get the response */
+	ret = inb(0x68);
+
+	/* Clear the "ownership flag" */
+	outb(0xFF, 0x6C);
+	return ret;
+}
+
 static void southbridge_init(struct device *dev)
 {
 	struct southbridge_amd_cs5536_config  *sb = (struct southbridge_amd_cs5536_config *)dev->chip_info;
@@ -107,13 +130,15 @@ static void southbridge_init(struct devi
 	setup_i8259(); 
 
 	if (sb->lpc_serirq_enable) {
+		int ret;
 		msr.lo = sb->lpc_serirq_enable;
-#if 0  // Not Yet
-#define EC_3700_SIGNATURE  0xa0
-                if (ec_inb(0xff00) == EC_3700_SIGNATURE) {
-                    msr.lo |= 0x40;  // Set quiet mode bit
-                }
-#endif
+
+		/* Get the HW version */
+		ret = eccmd(0x09);
+
+		if (ret != EC_3920_SIGNATURE)
+			msr.lo |= 0x40;
+
 		msr.hi  = 0;
 		wrmsr(MDD_LPC_SIRQ, msr);
 	}
