Index: src/mainboard/olpc/rev_a/auto.c
===================================================================
--- a/src/mainboard/olpc/rev_a/auto.c	(revision 2392)
+++ b/src/mainboard/olpc/rev_a/auto.c	(working copy)
@@ -122,10 +122,11 @@
 	/* the msr value reported by quanta is very, very different. 
 	 * we will go with that value for now. 
 	 */
-	msr.lo = 0x286332a3;
+	msr.lo = 0x686332a3;
 
 	wrmsr(0x20000019, msr);
 
+	print_err("CL2.5 Set\n");
 }
 
 #include "northbridge/amd/gx2/raminit.c"
