Index: LinuxBIOSv2/src/cpu/amd/model_gx2/cpubug.c
===================================================================
--- LinuxBIOSv2.orig/src/cpu/amd/model_gx2/cpubug.c	2006-08-29 08:44:01.000000000 -0600
+++ LinuxBIOSv2/src/cpu/amd/model_gx2/cpubug.c	2006-08-29 08:45:21.000000000 -0600
@@ -176,7 +176,7 @@
 	wrmsr(0x3003, msr);
 
 	/* change this value to zero if you need to disable this BTB SWAPSiF. */
-	if (1) { 
+	if (0) {
 
 		/* Disable enable_actions in DIAGCTL while setting up GLCP */
 		msr.hi = 0;
@@ -373,7 +373,8 @@
 			pcideadlock();
 			eng1398();
 			eng2900();
-			bug118339();
+			/* Disabled because it freaks out FS2 */
+			//bug118339();
 			break;
 		case 0x22:
 		case 0x30: 
